Week1/homework/world-queries.txt

```diff
@@ -1,40 +1,231 @@
+
+ +----------------------------+
+ | Name                       |
+ +----------------------------+
+ | Mumbai (Bombay)            |
+ | Seoul                      |
+ | São Paulo                  | 
+ | Shanghai                   |
+ | Jakarta                    |
+ | Karachi                    |
+ | Istanbul                   |
+ | Ciudad de México           |
+ | Moscow                     |
+ | New York                   |
+ +----------------------------+
+10 rows in set (0,01 sec)
+
+---------------------------------------------------------------------------------------------------
+
+6.Select the top 3countries by population that their names start with letter ‘P’
+
+SELECT Name FROM country WHERE Name LIKE "P%" ORDER BY Population DESC LIMIT 3 ;
+
+ +----------------------------+
+ | Name                       |
+ +----------------------------+
+ | Pakistan                   | 
+ | Philippines                |
+ | Poland                     |
+ +----------------------------+
+3 rows in set (0,00 sec)
+
+---------------------------------------------------------------------------------------------------
+
+
+7. What is the population in Asia?
+
+SELECT SUM(Population) FROM country WHERE Continent = "Asia" ;
+
+ +---------------------------+
+ | SUM(Population)           |
+ +---------------------------+
+ | 3705025700                |
+ +---------------------------+
+1 row in set (0,00 sec)
+
+---------------------------------------------------------------------------------------------------
+
+8. Which languages are spoken in the Region of ‘South America’
+
+ SELECT DISTINCT countrylanguage.language FROM countrylanguage JOIN country ON country.Code = countrylanguage.CountryCode WHERE country.Continent = "South America";
+
+ +--------------------------+
+ | language                 |
+ +--------------------------+
+ | Indian Languages         |
+ | Italian                  |
+ | Spanish                  |
+ | Aimará                   |
+ | Guaraní                  |
+ | Ketšua                   |
+ | German                   |
+ | Japanese                 |
+ | Portuguese               |
+ | Araucan                  |
+ | Rapa nui                 |
+ | Arawakan                 |
+ | Caribbean                |
+ | Chibcha                  | 
+ | Creole English           |
+ | English                  |
+ | Creole French            |
+ | Hindi                    | 
+ | Sranantonga              | 
+ | Goajiro                  |
+ | Warrau                   |
+ +--------------------------+
+21 rows in set (0,00 sec)
+
+---------------------------------------------------------------------------------------------------
+
+
+
+9. What are the languages spoken on all cities named ‘Barcelona’ (you may need to join 3 tables =) )
+
+SELECT DISTINCT language FROM countrylanguage WHERE CountryCode IN (SELECT code FROM country WHERE code In ( SELECT CountryCode FROM city WHERE Name = "Barcelona"));
```

Comment:
Ah, I've see you've used a triple subquery. This works okay, and will usually be optimised into a JOIN by the DBMS anyway, but it is, in theory, slower than a JOIN, and is also, in my opinion, less readable. I'd like it if you can try this one again using JOINs instead of subqueries. Let me know if you want some materials to read on JOINs.